src/main/java/com/minecolonies/coremod/colony/ColonyView.java

```diff
@@ -49,6 +53,36 @@
     private       int                                  maxCitizens = 0;
 
     /**
+     * Check if the colony has a warehouse.
+     */
+    private boolean hasWarehouse;
+
+    /**
+     * The Positions which players can freely interact.
+     */
+    private Set<BlockPos> freePositions = new HashSet<>();
+
+    /**
+     * The Blocks which players can freely interact with.
+     */
+    private Set<Block> freeBlocks = new HashSet<>();
+
+    /**
+     * The Set of waypoints.
+     */
+    private Set<BlockPos> wayPoints = new HashSet<>();
+
+    /**
+     * The overall happiness of the colony.
+     */
+    private double overallHappiness = 5;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)